### PR TITLE
[E2E]Push test fixes

### PR DESCRIPTION
**Pull request by pragnagopa**
- Fix push tests flakiness

### CODE REVIEW

**Comment by pragnagopa** [comment]
Fix for #21

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs [hunk 1]
```diff
@@ -154,37 +141,6 @@ public class PushFunctional : FunctionalTestBase
...
-        //    catch (MobileServiceInvalidOperationException)
-        //    {
-        //        throw;
-        //    }
-        //    finally
-        //    {
-        //        push.UnregisterAsync().Wait();
-        //    }
-        //}
-
```
Comment (danderson00) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs]: I think this test should be investigated before removal - we should understand whether it is an issue in NH or on our side that is causing the problem.

Comment (phvannor) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs]: I think RegisterAsyncTemplatesAndOverride is replacing this one.  Just unfortunate code lineup...  Looks like "RegisterAsyncWithTemplates" was found redundant instead.

Comment (pragnagopa) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs]: Yes. RegisterAsyncTemplatesAndOverride test covers verifying template registration and overriding registration without unregistration.

Comment (pragnagopa) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs]: @danderson00, @phvannor sign off?

**Comment by phvannor** [comment]
:+1:

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/Functional/PushFunctional.Test.cs [hunk 2]
```diff
@@ -185,9 +147,9 @@ private static JObject GetTemplates(string tag)
         {
             var dummyUser = await this.GetClient().InvokeApiAsync("JwtTokenGenerator", HttpMethod.Get, null);
 
-            MobileServiceUser user = new MobileServiceUser((string)dummyUser["token"]["payload"]["uid"])
+            MobileServiceUser user = new MobileServiceUser((string)dummyUser["userId"])
             {
-                MobileServiceAuthenticationToken = (string)dummyUser["token"]["rawData"]
+                MobileServiceAuthenticationToken = (string)dummyUser["autheticationToken"]
             };
             return user;
```
Comment (danderson00) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/Functional/PushFunctional.Test.cs]: This user stuff is duplicated in https://github.com/Azure/azure-mobile-apps-net-client/blob/master/e2etest/test/Microsoft.WindowsAzure.MobileServices.Test/TestData/Utilities.cs. It should probably reuse that implementation.

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs (hunk 1 above)
Comment (danderson00) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs]: Aside from comment on iOS, looks good!

Comment (pragnagopa) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.WindowsStore.Test/Functional/PushFunctional.Test.cs]: Fixed.

**Pull request merged**